### PR TITLE
Fi bo 24 initial django project

### PR DESCRIPTION
**Pull request by Jonas-Karl**

### CODE REVIEW

File: backend/api/models.py [hunk 1]
```diff
@@ -1,7 +1,5 @@
-import uuid
```
Comment (jens1o) [inline backend/api/models.py:1]: why was this removed? If I run `python manage.py`, I get an error because `uuid.uuid4` cannot be found :)

Comment (Jonas-Karl) [reply backend/api/models.py:1]: Actually uuid.uuid4 should not be used anymore

Comment (jens1o) [reply backend/api/models.py:1]: Still, I get the error upon launch (`python manage.py` does not work as well)?

Comment (Jonas-Karl) [reply backend/api/models.py:1]: Maybe you still got an old database scheme that somewhere uses uuid4?

Comment (jens1o) [reply backend/api/models.py:1]: that did the trick, thank you!

**Comment by jens1o** [comment]
```
<h1>ValueError
       at /users/register/</h1>
  <pre class="exception_value">Couldn't load 'Argon2PasswordHasher' algorithm library: No module named 'argon2'</pre>
  

Request Method: | POST
-- | --
http://127.0.0.1:8000/users/register/
4.1.3
ValueError
Couldn't load 'Argon2PasswordHasher' algorithm library: No module named 'argon2'
/Users/Jens.Hausdorf/.pyenv/versions/3.11.2/lib/python3.11/site-packages/django/contrib/auth/hashers.py, line 205, in _load_library
api.views.registerUser
/Users/Jens.Hausdorf/.pyenv/versions/3.11.2/bin/python
3.11.2
['/Users/Jens.Hausdorf/dev/FiBo/backend',  '/Users/Jens.Hausdorf/.pyenv/versions/3.11.2/lib/python311.zip',  '/Users/Jens.Hausdorf/.pyenv/versions/3.11.2/lib/python3.11',  '/Users/Jens.Hausdorf/.pyenv/versions/3.11.2/lib/python3.11/lib-dynload',  '/Users/Jens.Hausdorf/.pyenv/versions/3.11.2/lib/python3.11/site-packages']
Thu, 06 Apr 2023 19:05:21 +0000
```
sorry I'm keeping blaming you ;D

but it looks like there is an entry to the requirements.txt file missing? :)

File: backend/api/views.py [hunk 2]
```diff
@@ -1,8 +1,90 @@
...
+@api_view(['POST'])
+def authenticateUser(request):
+    # Note that username and email are the same
+    user = authenticate(username=request.data['email'], password=request.data['password'])
+    if user is not None:
+        # TODO login(request,user)
+        # redirect to a success page
+        return Response(f'Authenticated user {user.email}')
+    else:
+        return Response('Invalid login')
```
Comment (jens1o) [inline backend/api/views.py:46]: I think this should use http status codes for better testability.

If a user could authenticate correctly, it should send 200, otherwise 401 (see https://stackoverflow.com/a/45357230)

File: backend/api/views.py [hunk 3]
```diff
@@ -1,8 +1,90 @@
...
 
-def index(request):
-    return HttpResponse("Hello, world. You're at the api index.")
\ No newline at end of file
+
+@api_view(['GET'])
+def getUser(request, pk):
+    user = FiboUser.objects.get(id=pk)
+    serializer = FiboUserSerializer(user, many=False)
+    return Response(serializer.data)
```
Comment (jens1o) [inline backend/api/views.py:19]: we should not allow _unauthenticated_ users of the API to acess whatever private, personal-identifiable, data we have saved in our database. :D

Could you please either disable this API endpoint or make sure it does not return any data for unauthenticated users?

**Review by jens1o — CHANGES_REQUESTED** [review]
looks good and works on my machine.

Thank you so much for your work, Markus and Jonas. I've left a couple of comments I'd like to see adressed prior to merging.

File: backend/api/views.py [hunk 4]
```diff
@@ -1,8 +1,77 @@
...
+from django.contrib.auth import authenticate
 
-# Create your views here.
-from django.http import HttpResponse
 
+@api_view(['GET'])
+def getUser(request, pk):
+    user = FiboUser.objects.get(id=pk)
+    serializer = FiboUserSerializer(user, many=False)
+    return Response(serializer.data, status=status.HTTP_200_OK)
```
Comment (jens1o) [inline backend/api/views.py:13]: I've created a [backlog entry](https://dhbw-karlsruhe.myjetbrains.com/youtrack/issue/FiBo-161/Put-users-id-behind-authentication-authorization) to make sure we only allow to return [PII] when authenticated + authorized.

[PII]: https://en.wikipedia.org/wiki/Personal_data

**Review by jens1o — APPROVED** [review]
Thank you! All my concerns were adressed and since nobody else reviewed or at least self-requested a review, I'll merge.
